inject-java/src/test/groovy/io/micronaut/inject/requires/RequiresSpec.groovy

```diff
@@ -384,6 +384,40 @@ class MyBean {
         context.close()
     }
 
+    void "test requires missing classes as string class name when class present"() {
+        when:
+        BeanDefinition beanDefinition = buildBeanDefinition('test.MyBean', '''
+package test;
+
+import io.micronaut.context.annotation.*;
+
```

Comment:
Seems like there is an issue in micronaut-core 4.6.x when declaring something like this 
`@Requires(missingClasses = "org.hibernate.reactive.provider.ReactiveServiceRegistryBuilder")` because this code in `MatchesConditionUtils` 
```
if (requirement.contains(RequiresCondition.MEMBER_MISSING_CLASSES)) {
            AnnotationClassValue<?>[] classes = requirement.annotationClassValues(RequiresCondition.MEMBER_MISSING_CLASSES);
            if (classes.length > 0) {
                preConditions.add(new MatchesAbsenceOfClassesCondition(classes));
            }
        }
```
expects to get classes in annotation value (but can't if class not on the classpath?)
```
public AnnotationClassValue<?>[] annotationClassValues(@NonNull String member) {
        if (StringUtils.isEmpty(member)) {
            return AnnotationClassValue.ZERO_ANNOTATION_CLASS_VALUES;
        }
        Object o = values.get(member);
        if (o instanceof AnnotationClassValue<?> annotationClassValue) {
            return new AnnotationClassValue[]{annotationClassValue};
        }
        if (o instanceof AnnotationClassValue<?>[] annotationClassValues) {
            return annotationClassValues;
        }
        return AnnotationClassValue.ZERO_ANNOTATION_CLASS_VALUES;
    }
```
in this case, instanceof returns false since value is string and then fails to match condition.